src/lib/parser.rs

```diff
@@ -150,7 +150,7 @@ impl Parser {
       }
     }
 
-    let mut char_count: usize = 0;
+    let mut char_count: usize = 1;
```

Comment:
This won't work because a few lines after here we do `match option.as_bytes().get(char_count) {` and that will return invalid relative characters.  
below the `while` we can better do:
```
self.index -= char_count + 1;
```

Reply:
Ah, this is much better. I'll make the changes now